kb(grid): customize delete confirmation dialog (#1173)
telerik/blazor-docs: knowledge-base/grid-customize-delete-confirmation-dialog.md

---
title: Customize the Delete Confirmation Dialog
description: How to customize the Delete Confirmation Dialog? 
type: how-to
page_title: Customize the Delete Confirmation Dialogs
slug: grid-kb-customize-delete-confirmation-dialog
position: 
tags: grid, customize, delete, confirmation, dialog, message
ticketid: 1553006
res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>

> 			<td rowspan = 3>Product</td>

Reviewer: Why not use all component names in a single cell with `<br />`s in between?

			<td>Grid for Blazor</td>
		</tr>
		<tr>
			<td>TreeList for Blazor</td>
		</tr>
		<tr>
			<td>Scheduler for Blazor</td>
		</tr>				
	</tbody>
</table>

## Description

How to customize the built-in Delete Confirmation Dialog of the Grid?

I want to add some item details to the text of the Delete Confirmation Dialog to notify the user for the item they are trying to delete. How to achieve that?

How to change the text of the buttons in the Delete Confirmation Dialog?

How to change the title and the content of the Delete Confirmation Dialog?

## Solution

> The built-in Delete Confirmation Dialog is designed to be enabled with minimum effort and code (only by enabling the `ConfirmDelete` parameter of the Grid). Thus, the dialog itself does not provide customization options. However, there are several approaches that you can aplly to achieve a custom Delete Confirmation Dialog.

Reviewer: ```suggestion
The built-in Delete Confirmation Dialog requires minimum effort and code. It only needs the `ConfirmDelete` parameter of the Grid. Thus, the dialog itself does not provide customization options. However, there are several ways to achieve a custom Delete Confirmation Dialog.
```

### Localization

You can enable [Localization]({%slug globalization-localization%}) in your application and provide the desired custom text for the corresponding element keys.

The keys for the elements of the built-in Delete Confirmation Dialog are:

| Element | Key |
| --- | --- |
| Title | `Grid_ConfirmDeleteTitle` |
| Text | `Grid_ConfirmDeleteText` |
| OK button | `Grid_ConfirmDeleteOk` |
| Cancel button | `Grid_ConfirmDeleteCancel` |

>tip The keys for the Treelist and Scheduler components are the same, just change the component name before the `_` symbol.

### Predefined Dialog Component

Use [Predefined Confirm Dialog]({%slug dialog-predefined%}#confirm) with the desired custom text. Additionally, you may get the details for the current item and add them to the text:

* Handle the [`OnDelete`]({%slug grid-events%}#cud-events) event of the Grid
* Display the Predefined Dialog in the `OnDelete` handler
* Cancel the event or proceed with the `OnDelete` logic depending on the user choice

````CSHTML

> <TelerikGrid Data=@GridData

Reviewer: We don't need full-blown CRUD examples. The code snippets will be shorter and simpler if they only use Delete commands (no editing and no adding).
